toml/src/main/kotlin/org/rust/toml/crates/local/completion/DependencyCratesLocalCompletionProvider.kt

```diff
@@ -0,0 +1,78 @@
+import com.intellij.codeInsight.lookup.LookupElementPresentation
+import com.intellij.codeInsight.lookup.LookupElementRenderer
+import com.intellij.icons.AllIcons
+import com.intellij.openapi.components.service
+import org.rust.toml.StringValueInsertionHandler
+import org.rust.toml.completion.CargoNormalizedNamesPrefixMatcher
+import org.rust.toml.completion.TomlKeyValueCompletionProviderBase
+import org.rust.toml.crates.local.CratesLocalIndexService
+import org.rust.toml.crates.local.lastVersion
+import org.toml.lang.psi.TomlKeyValue
+
+class DependencyCratesLocalCompletionProvider : TomlKeyValueCompletionProviderBase() {
+    override fun completeKey(keyValue: TomlKeyValue, result: CompletionResultSet) {
+        val prefix = CompletionUtil.getOriginalElement(keyValue.key)?.text ?: return
+
+        val indexService = service<CratesLocalIndexService>()
+        if (!indexService.isReady) return
+
+        val crateNames = indexService.getAllCrateNames()
+        val elements = crateNames.mapNotNull { crateName ->
+            val crate = indexService.getCrate(crateName) ?: return@mapNotNull null
+
+            PrioritizedLookupElement.withPriority(
+                LookupElementBuilder
+                    .create(crateName)
+                    .withIcon(AllIcons.Nodes.PpLib)
+                    .withExpensiveRenderer(object : LookupElementRenderer<LookupElement>() {
+                        override fun renderElement(element: LookupElement?, presentation: LookupElementPresentation) {
+                            presentation.itemText = "$crateName = \"${crate.lastVersion ?: ""}\""
+                        }
+                    })
+                    .withInsertHandler { context, _ ->
+                        context.document.insertString(context.tailOffset, " = \"${crate.lastVersion}\"")
+                        val endLineOffset = context.editor.caretModel.visualLineEnd
+                        // TODO: Currently moves caret to the next line
+                        context.editor.caretModel.moveToOffset(endLineOffset)
+                    },
+                (-crateName.length).toDouble()
+            )
+        }
+        result.withPrefixMatcher(CargoNormalizedNamesPrefixMatcher(prefix)).addAllElements(elements)
+    }
+
+    override fun completeValue(keyValue: TomlKeyValue, result: CompletionResultSet) {
+        val name = CompletionUtil.getOriginalElement(keyValue.key)?.text ?: return
+
+        val indexService = service<CratesLocalIndexService>()
+        if (!indexService.isReady) return
+
+        val versions = indexService.getCrate(name)?.versions ?: return
+        val elements = versions.mapIndexed { index, variant ->
+            var lookupElement = LookupElementBuilder.create(variant.version)
+                .withInsertHandler(StringValueInsertionHandler(keyValue))
+
+            if (variant.isYanked) {
+                lookupElement = lookupElement.withTailText(" yanked")
+            }
```

Comment:
```suggestion
            val lookupElement = LookupElementBuilder.create(variant.version)
                .withInsertHandler(StringValueInsertionHandler(keyValue))
                .withTailText(if (variant.isYanked) " yanked" else null)
```

Looks like this works fine as well